grading/main_validator.cpp

```diff
@@ -397,13 +398,106 @@ void ValidateATestCase(nlohmann::json config_json, int which_testcase,
     testcase_message,
     testcase_pts,
     all_testcases);
+    WriteToGradefile(my_testcase,gradefile,testcase_pts);
+}
+
+/**
+* Get all testcase objects associated with an assignment.
+*/
+std::vector<TestCase> getAllTestcases(nlohmann::json config_json) {
+  std::vector<TestCase> testcase_array;
+  nlohmann::json::const_iterator testcases = config_json.find("testcases");
+  assert (testcases != config_json.end());
+  for (nlohmann::json::const_iterator tc = testcases->begin(); tc != testcases->end(); tc++) {
+    std::string testcase_id = tc->value("testcase_id", "");
+    assert(testcase_id != "");
+    nlohmann::json testcase_config = *tc;
+    TestCase my_testcase(testcase_config, testcase_id, "");
+    testcase_array.push_back(my_testcase);
+  }
+
+  nlohmann::json::const_iterator item_pool = config_json.find("item_pool");
+  if (item_pool != config_json.end()){
+    for(nlohmann::json::const_iterator item_itr = item_pool->begin(); item_itr != item_pool->end(); item_itr++) {
+      nlohmann::json item_testcases = item_itr->value("testcases", nlohmann::json::array());
+      for(nlohmann::json::const_iterator it_itr = item_testcases.begin(); it_itr != item_testcases.end(); it_itr++) {
+        std::string testcase_id = it_itr->value("testcase_id", "");
+        assert(testcase_id != "");
+        nlohmann::json testcase_config = *it_itr;
+        TestCase my_testcase(testcase_config, testcase_id, "");
+        testcase_array.push_back(my_testcase);
+      }
+    }
+  }
 
-    WriteToGradefile(which_testcase,my_testcase,gradefile,testcase_pts);
+  return testcase_array;
+}
+
+nlohmann::json getItemWithName(std::string item_name, const nlohmann::json& config_json) {
+
+    nlohmann::json::const_iterator itempool = config_json.find("item_pool");
+    for (nlohmann::json::const_iterator item = itempool->begin(); item != itempool->end(); item++) {
+        if((*item)["item_name"] == item_name) {
+          return *item;
+        }
+    }
+    throw "ERROR: Could not find an item with item_name " + item_name;
+}
+
+
+/**
+* Get all global testcase objects for an assignment + all testcases specified in .submit.notebook
```

Comment:
Seems to overlap alot in functionality with getalltestcases
(and other functions in other files?)

library / standardize naming, 
use same function with additional args to do different things?

Reply:
I'll type up an issue about making an  AutogradingConfig class in the cpp code.